### PR TITLE
docs(examples): centralize styles

### PR DESCRIPTION
**Pull request by jul-lam**
## 💡 What is the current behavior?

The appearance of the preview examples lacks consistency and varies across different frameworks.

JIRA Ticket: **IX-1118**
GitHub Issue Number: X

## 🆕 What is the new behavior?

Styling is now managed through CSS files located in a newly created central package. Logic has been incorporated to optionally display these files in the code preview. Additionally, minor issues in the code preview appearance have been fixed.

## 🏁 Checklist

A pull request can only be merged if all of these conditions are met (where applicable):

- [x] 🦮 Accessibility (a11y) features were implemented
- [x] 🗺️ Internationalization (i18n) - no hard coded strings
- [x] 📲 Responsiveness - components handle viewport changes and content overflow gracefully
- [x] 📄 Documentation was reviewed/updated (`pnpm run docs`)
- [x] 🧪 Unit tests were added/updated and pass (`pnpm test`)
- [x] 📸 Visual regression tests were added/updated and pass ([Guide](https://github.com/siemens/ix/blob/main/CONTRIBUTING.md#visual-regression-testing))
- [x] 🧐 Static code analysis passes (`pnpm lint`)
- [x] 🏗️ Successful compilation (`pnpm build`, changes pushed)

## 👨‍💻 Help & support

X

### CODE REVIEW

File: packages/documentation/scripts/generate-markdown.ts [hunk 1]
```diff
@@ -177,27 +182,35 @@ const tasks = new Listr<Context>(
     {
       title: 'Rename code snippets',
       task: async () => {
+        const renameFilesInDirectory = async (
+          directory: string
+        ): Promise<any[]> => {
```
Comment (danielleroux) [inline packages/documentation/scripts/generate-markdown.ts:187]: Make correct time avoid any here

File: packages/documentation/scripts/generate-markdown.ts [hunk 2]
```diff
@@ -177,27 +182,35 @@ const tasks = new Listr<Context>(
...
+            entries.flatMap((entry) => {
+              const entryPath = path.join(directory, entry.name);
+
+              if (entry.isDirectory()) {
+                return renameFilesInDirectory(entryPath);
+              }
+
+              return fs.rename(
+                entryPath,
+                path.join(directory, `${entry.name}.txt`)
```
Comment (danielleroux) [inline packages/documentation/scripts/generate-markdown.ts:200]: Why is the .txt postfix necessary?

File: packages/documentation/src/components/PlaygroundV2/utils.ts [hunk 3]
```diff
@@ -86,18 +102,32 @@ async function openHtmlStackBlitz(
   sourceFiles: SourceFile[],
   version: string
 ) {
-  const [main_js, package_json, vite_config_ts] =
+  const styleFilePath: string = `${baseUrl}auto-generated/previews/styles/`;
+
+  const [global_css, main_js, package_json, vite_config_ts] =
     await loadSourceCodeFromStatic([
+      `${styleFilePath}global.css.txt`,
```
Comment (danielleroux) [inline packages/documentation/src/components/PlaygroundV2/utils.ts:109]: Result of the .txt postfix. Looks odd to have .txt file here

File: packages/documentation/src/components/PlaygroundV2/utils.ts [hunk 4]
```diff
@@ -249,14 +323,16 @@ async function openReactStackBlitz(
...
+        'public/index.html': index_html,
         'src/index.tsx': index_tsx,
         'src/App.tsx': patchAppTs(),
         'package.json': patchPkgLibraryVersion(package_json, version),
         'tsconfig.json': tsconfig_json,
         '.stackblitzrc': `{
-        "startCommand": "yarn run start"
-      }`,
+          "startCommand": "yarn run start"
+        }`,
```
Comment (danielleroux) [inline packages/documentation/src/components/PlaygroundV2/utils.ts:335]: dont use `yarn`

**Review by danielleroux — CHANGES_REQUESTED** [review]

**Review by danielleroux — APPROVED** [review]

File: packages/documentation/src/components/PlaygroundV2/utils.ts [hunk 5]
```diff
@@ -176,30 +206,52 @@ async function openAngularStackBlitz(
...
+      );
+      source = adaptedSource;
+
+      // get style file
+      if (styleFileName) {
+        styleFiles[`src/app/${styleFileName}`] = (
+          await loadSourceCodeFromStatic([`${styleFilePath}${styleFileName}`])
+        )[0];
+      }
+    }
```
Comment (nuke-ellington) [inline packages/documentation/src/components/PlaygroundV2/utils.ts:231]: This code block is essentially duplicated thrice in this file. Maybe we can extract it with the file suffixes as parameters?

**Review by nuke-ellington — CHANGES_REQUESTED** [review]

**Review by nuke-ellington — APPROVED** [review]

**Pull request merged**